example/reinforcement-learning/dqn/base.py

```diff
@@ -135,7 +135,7 @@ def switch_bucket(self, bucket_kwargs=None, data_shapes=None):
                 self.initializer(k, v)
         else:
             assert set(arg_name_shape.items()) == \
-                   set(data_shapes.items() + [(k, v.shape) for k, v in self.params.items()])
+                   set(list(data_shapes.items()) + list([(k, v.shape) for k, v in self.params.items()]))
```

Comment:
In python3.x, this can also be fixed by changing to union of dict_items as -
`set(data_shapes.items() | dict([(k, v.shape) for k, v in self.params.items()]).items())`

However, this fails in python2.7 as this results in a union of lists.

Reply:
This fix looks good.